lib/parser.js

```diff
@@ -613,7 +613,15 @@ let whileLoop = seqMap(keywords.while,
 let ifElse = seqMap(keywords.if,
   expr,
   block,
-  (keywords.else).then(block).times(0, 1).map((elses) => {
+  (keywords.else).then(alt(block, lazy(() => {
+    return ifElse;
+  }).source().map((elseIf) =>({
```

Comment:
Sure, I can pull the `.source().map()` into the `lazy` block.  I'm not sure I can omit the `lazy` though since `ifElse` is not yet defined at this point (we are in the middle of defining it).